src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs

```diff
@@ -1059,6 +1069,39 @@ public static FormulaValue OptionSetValueToLogicalName(IRContext irContext, Opti
             var optionSet = args[0];
             var logicalName = optionSet.Option;
             return new StringValue(irContext, logicalName);
+        }
+
+        public static FormulaValue PlainText(IRContext irContext, FormulaValue[] args)
+        {
+            if (args[0].Type == FormulaType.String)
```

Comment:
>if (args[0].Type == FormulaType.String) [](http://example.com/codeflow?start=12&length=39)

Can we avoid this check and use error handling pipeline maybe?

Reply:
this is not error handling. If it is string format then I tried to decode it.

Reply:
When would this be not string type? (Wouldn't Checktype already catch it?)

Reply:
i just allow they input differnt format here: string, blank() or datetime or boolean. but the decoding only process for string.

Reply:
Since this function declares its argument as string, it will only receive a string - if we pass a different data type it will be first coerced to a string prior to being passed to the function. You can check on the REPL project to see what happens under the hood.

For example, if you enter `IR(Len(Today()))` in the repl console, you'll see the call to the Text function (which implements Date[Time]-to-text coercion) in the IR:

    Len:w(
        BlankToEmptyString:s(
            Text:s(Today:D(), "'shortdate'":s)))

Reply:
I see. thanks for explaining. i changed this back to stringvalue set.